synapse/storage/registration.py

```diff
@@ -860,18 +860,18 @@ def _background_update_set_deactivated_flag_txn(txn):
             )
 
             if batch_size > len(rows):
-                return True
+                return (True, rows_processed_nb)
```

Comment:
You probably want to use the number of rows fetched from the DB (i.e. `len(rows)`), as that's what we're limiting against